src/database/postgresPersistence.ts

```diff
@@ -65,6 +73,7 @@ export class PostgresPersistance implements PersistanceObject {
             'eventName',
             'args',
             'chainId',
+            'transactionHash',
           )} ON CONFLICT(id) DO UPDATE SET "blockNumber"=excluded."blockNumber"`;
```

Comment:
When conflict occurs, this means there were a chain reorg. In that case, we'd also need to swap the transactionHash because it will be different than it was before.